### PR TITLE
Repo: update readme narrative information and other details

### LINKED ISSUE #1: Complete README

### PR DESCRIPTION
**Pull request by jasonklein**
This PR updates the readme's narrative information and installation and usage details.

It largely tracks the [OpenST readme](https://github.com/OpenST/openst-contracts), but there are some notable exceptions:
- the OpenST readme says it uses Ethlint a/k/a solium. This repo has solium config files, but has neither ethlint nor solium in its `package.json`. Consequently, this is not mentioned in the readme.

✏️When a solidity linting npm package is added, updating the readme to reflect the linter should be part of that ticket (although perhaps under a larger story of aligning the dev tools and scripts)
- the OpenST readme instructs adding `./node_modules/.bin` to the PATH. There was an internal discussion about relying on locally-installed NPM packages rather than globally. There appears to have been a consensus in favor of this approach, but it was only adopted in `openst-contracts`.

✏️ If the scripts in this readme will be updated to use locally-installed packages, that ticket should also include updating the readme to include this instruction.

Resolves #1.

### CODE REVIEW

**Comment by schemar** [comment]
On it 🔍

File: README.md [hunk 1]
```diff
@@ -2,56 +2,104 @@
...
 cd organization-contracts
 npm install
+```
+## Usage
+
+The contracts and interfaces in this repository are primarily intended for use in other OpenST projects. Please see [Related Work].
+
+### Compiling the contracts
+
+The following npm script compiles updated contracts from the last call using [truffle]:
```
Comment (schemar) [inline README.md]: "from the last call"?

File: README.md [hunk 2]
```diff
@@ -2,56 +2,104 @@
...
+
+```bash
+npm run compile
+```
+
+To compile all contracts:
```
Comment (schemar) [inline README.md]: Some more explanation as to why it is relevant maybe?

```suggestion
To compile all contracts, including the ones that haven't changed:
```

File: README.md [hunk 3]
```diff
@@ -2,56 +2,104 @@
...
+We use the [ganache] blockchain for development. Before running the tests, start `ganache`:
+
+```bash
+npm run ganache-cli
+```
+
+Test the contracts using [truffle]:
+
+```bash
 npm run test
```
Comment (schemar) [inline README.md]:
```suggestion
npm test
```

File: README.md [hunk 4]
```diff
@@ -2,56 +2,104 @@
...
 We track our [issues] on GitHub.
 
-To contribute code, please ensure that your read the [developer guidelines] first.
+To contribute code, please review our [Developer Guidelines] and ensure that your submissions adhere to the [Style Guide]; please also be aware, this project is under active development and we have not yet established firm contribution guidelines or acceptance criteria.
 
 ### Community
 
 * [Forum][discourse]
-* [Gitter]
+* [Gitter][gitter]
```
Comment (schemar) [inline README.md]: Is this change required?

**Review by schemar — CHANGES_REQUESTED** [review]
Looks almost good to me! 🐨
Just a few questions and comments.

> * the OpenST readme instructs adding `./node_modules/.bin` to the PATH. There was an internal discussion about relying on locally-installed NPM packages rather than globally. There appears to have been a consensus in favor of this approach, but it was only adopted in `openst-contracts`.
> 
> 
> ✏️ If the scripts in this readme will be updated to use locally-installed packages, that ticket should also include updating the readme to include this instruction.

While I strongly agree that locally installed tools should be used instead of globally installed tools, I strongly disagree to force anyone to set-up their local environment (globally) to use my specific repository. NPM `run` scripts already use local versions of installed tools by default.
From the [npm documentation]:
> In addition to the shell’s pre-existing PATH, npm run adds node_modules/.bin to the PATH provided to scripts. Any binaries provided by locally-installed dependencies can be used without the node_modules/.bin prefix. For example, if there is a devDependency on tap in your package, you should write:
>
> `"scripts": {"test": "tap test/\*.js"}`
>
> instead of
>
> `"scripts": {"test": "node_modules/.bin/tap test/\*.js"}`
>
> to run your tests.

[npm documentation]: https://docs.npmjs.com/cli/run-script#description

**Comment by pgev** [comment]
> Looks almost good to me! 🐨
> Just a few questions and comments.
> 
> > * the OpenST readme instructs adding `./node_modules/.bin` to the PATH. There was an internal discussion about relying on locally-installed NPM packages rather than globally. There appears to have been a consensus in favor of this approach, but it was only adopted in `openst-contracts`.
> > 
> > ✏️ If the scripts in this readme will be updated to use locally-installed packages, that ticket should also include updating the readme to include this instruction.
> 
> While I strongly agree that locally installed tools should be used instead of globally installed tools, I strongly disagree to force anyone to set-up their local environment (globally) to use my specific repository. NPM `run` scripts already use local versions of installed tools by default.
> From the [npm documentation](https://docs.npmjs.com/cli/run-script#description):

I've advocated to add `./node_modules/.bin` to the PATH variable. However, I see and agree with Martin's point that having it in developer guideline (especially for external contributors) are confusing. Rather, we should aim always write write scripts/commands in a way that they refer to locally installed npm packages.

Here is an `./tools/run_ganache_cli.sh` script from `openst-contracts`

```bash
#!/bin/bash

script_dir_path="$(cd "$(dirname "$0")" && pwd)"

"${script_dir_path}/../node_modules/.bin/ganache-cli" \
    --accounts=100 \
    --defaultBalanceEther=100 \
    --gasLimit 0xfffffffffff
```

It's explicitly using the locally installed `ganache-cli` module.


> 
> > In addition to the shell’s pre-existing PATH, npm run adds node_modules/.bin to the PATH provided to scripts. Any binaries provided by locally-installed dependencies can be used without the node_modules/.bin prefix. For example, if there is a devDependency on tap in your package, you should write:
> > `"scripts": {"test": "tap test/\*.js"}`
> > instead of
> > `"scripts": {"test": "node_modules/.bin/tap test/\*.js"}`
> > to run your tests.

File: README.md (hunk 1 above)
Comment (jasonklein) [reply README.md]: "from the last call" --> "since the last compile" (will change!)

**Comment by jasonklein** [comment]
> I've advocated to add ./node_modules/.bin to the PATH variable. However, I see and agree with Martin's point that having it in developer guideline (especially for external contributors) are confusing. Rather, we should aim always write write scripts/commands in a way that they refer to locally installed npm packages.

@pgev: I'm having trouble squaring the above, with what @schemar noted from the NPM docs:
> In addition to the shell’s pre-existing PATH, npm run adds node_modules/.bin to the PATH provided to scripts.

Can you help me to understand?

**Comment by schemar** [comment]
@jasonklein sometimes scripts are run without using `npm run` like in the example that @pgev gave. We should make sure that we always use `npm run` (or alternatively the relative path to the node modules bin) to make sure that we always use the local versions of all executables.

**Comment by pgev** [comment]
> @jasonklein sometimes scripts are run without using `npm run` like in the example that @pgev gave. We should make sure that we always use `npm run` (or alternatively the relative path to the node modules bin) to make sure that we always use the local versions of all executables.

@schemar I've made a quick test and npm run does not add the current path to the PATH variable before running. Hence, we should make sure in our scripts to refer to local node_modules executables.

**Review by schemar — APPROVED** [review]
:shipit:
